rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -845,6 +836,16 @@ def train(self, training_data: TrainingData) -> Resource:
                 f"Skipping training of the classifier."
             )
             return self._resource
+
+        if not self.model and self.finetune_mode:
```

Comment:
changed that in alignment with @dakshvar22 (see Slack thread [here](https://rasa-hq.slack.com/archives/C8LEHMTAM/p1632297307058700)). Otherwise we were running into errors when finetuning a model which has a response selector but no retrieval intents. This further avoids training a ResponseSelector during finetuning when there were no retrieval intents previously